### PR TITLE
[docs] Migrate Output API and Output functions

### LINKED ISSUE #179: [docs] Migrate Output_API
### The URL of the page on https://docs.moodle.org/dev/

Output_API

### Is this documentation specific to a Moodle version?

Unsure (Default)

### What location would you suggest in the new docs?

/docs/category/subsystems/output.md

Apart from updating the content on this page, maybe it would be good to migrate to this Output API page the content in https://docs.moodle.org/dev/Output_functions

### PR DESCRIPTION
**Pull request by kabalin**
Migrated pages:
- https://docs.moodle.org/dev/Output_API
- https://docs.moodle.org/dev/Output_functions

Closes #179

<a href="https://gitpod.io/#https://github.com/moodle/devdocs/pull/216"></a>

### CODE REVIEW

File: docs/apis/subsystems/output.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
+---
+title: Output API
+tags:
+  - Output
+  - API
+---
+
+<!-- cspell:ignore renderables, sometext, courseid, enabletrusttext -->
+
+## Overview
```
Comment (andrewnicols) [inline docs/apis/subsystems/output.md:10]: I'd recommend taking this out. We've standardised to remove the first title if it's something like "Introduction", or "Overview". There is no benefit.

File: docs/apis/subsystems/output.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+$output = $PAGE->get_renderer('tool_demo');
+
+echo $output->header();
+echo $output->heading($pagetitle);
+
+$renderable = new \tool_demo\output\index_page('Some text');
+echo $output->render($renderable);
+
+echo $output->footer();
+
```
Comment (andrewnicols) [inline docs/apis/subsystems/output.md:47]:
```suggestion
```

File: docs/apis/subsystems/output.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+$PAGE->set_url($url);
+$PAGE->set_title($title);
+$PAGE->set_heading($title);
+```
+
+:::note What is $PAGE and where did it come from ?
+
+`$PAGE` is a global variable used to track the state of the page that is being returned. It is an instance of the `moodle_page` class defined in `lib/pagelib.php`. See [Page API](https://docs.moodle.org/dev/Page_API) for more information on the `$PAGE` variable.
+
+:::
```
Comment (andrewnicols) [inline docs/apis/subsystems/output.md:75]: This is largely irrelevant to the Output API docs.

I'd suggest putting it into a collapsed `<details>` section or an `:::info` section, for example:

```suggestion
:::info Setup of an admin page

On admin pages the `admin_externalpage_setup($sectionname);` function should be used to setup and perform permission checks, for example:

```php title="admin/tool/demo/mypage.php"
require_once(__DIR__ . '/../../config.php');
require_once("{$CFG->libdir}/adminlib.php");

admin_externalpage_setup('example');
```

:::

```

I'd suggest detailing the PAGE API still, but with more info.

**Review by andrewnicols — CHANGES_REQUESTED** [review]
I don't have a chance to finish a review now as I'm done for the day, but I have a couple of notes so far.

**Comment by andrewnicols** [comment]
Oh, I meant to say, once you've rebasd you'll need to run:

```yarn mdfix-all ; yarn mdxfix-all```

That will update any reference to the migrated docs. It will become a part of the mgiration script when #207 lands but for now it's a manual process.

**Comment by kabalin** [comment]
Thanks @andrewnicols , I addressed all the comments and fixed the cross-links. Not sure if I got your suggestion about `PAGE` correctly, please comment if more changes are needed.

**Review by andrewnicols — APPROVED** [review]
Looks good to me.

There's always going to be improvements that we can make here, but I think that they can happen _after_ we get documents migrated now.

**Pull request merged**